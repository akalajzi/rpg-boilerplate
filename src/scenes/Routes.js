// @flow strict
import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import ErrorBoundary from 'components/ErrorBoundary';
import { HomePage, BadUrlPage } from './index';

const Routes = () => (
  <BrowserRouter>
    <ErrorBoundary>
      <Switch>
        <Route exact path="/" component={HomePage} />
        {/* no match */}
        <Route component={BadUrlPage} />
      </Switch>
    </ErrorBoundary>
  </BrowserRouter>
);

export default Routes;
