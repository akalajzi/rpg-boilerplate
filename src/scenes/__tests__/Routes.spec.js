// @flow strict
import React from 'react';
import { shallow } from 'enzyme';

import Routes from '../Routes';

describe('#Routes', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Routes />);

    expect(wrapper.getElement()).toMatchSnapshot();
  });
});
