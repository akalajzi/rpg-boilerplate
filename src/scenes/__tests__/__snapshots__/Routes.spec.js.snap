// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Routes should render correctly 1`] = `
<BrowserRouter>
  <ErrorBoundary>
    <HelmetWrapper
      defer={true}
      encodeSpecialCharacters={true}
    >
      <title>
        RPG
      </title>
    </HelmetWrapper>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </ErrorBoundary>
</BrowserRouter>
`;
